plugins/woocommerce/client/blocks/assets/js/blocks/product-elements/add-to-cart-form/frontend.ts

```diff
@@ -49,17 +96,45 @@ const dispatchChangeEvent = ( inputElement: HTMLInputElement ) => {
 };
 
 store( 'woocommerce/add-to-cart-form', {
-	state: {},
+	state: {
+		get allowsDecrease() {
+			const { constraints, currentQuantity } = getQuantityStateInfo();
+			const { min: minValue, step } = constraints;
+			return currentQuantity - step >= minValue;
+		},
+		get allowsIncrease() {
+			const { constraints, currentQuantity } = getQuantityStateInfo();
+			const { max: maxValue, step } = constraints;
+			return maxValue === null || currentQuantity + step <= maxValue;
+		},
+	},
 	actions: {
 		addQuantity: ( event: HTMLElementEvent< HTMLButtonElement > ) => {
 			const inputData = getInputData( event );
 			if ( ! inputData ) {
 				return;
 			}
-			const { currentValue, maxValue, step, inputElement } = inputData;
+			const context = getContext< Context >();
+			const {
+				currentValue,
+				maxValue,
+				step,
+				inputElement,
+				childProductId,
+			} = inputData;
 			const newValue = currentValue + step;
-
-			if ( maxValue === undefined || newValue <= maxValue ) {
+			if ( maxValue === null || newValue <= maxValue ) {
+				if ( childProductId ) {
+					context.quantity = {
+						...context.quantity,
+						[ childProductId ]: newValue,
+					};
+				} else {
+					context.quantity = {
+						...context.quantity,
+						[ context.productId ]: newValue,
+					};
+				}
```

Comment:
I think this can be simplified as well:

```suggestion
				const id = childProductId || context.productId;
				context.quantity = {
					...context.quantity,
					[ id ]: newValue,
				};
```